apps/src/p5lab/spritelab/blocks.js

```diff
@@ -8,29 +8,51 @@ import {APP_HEIGHT, P5LabInterfaceMode} from '../constants';
 import {TOOLBOX_EDIT_MODE} from '../../constants';
 import {animationSourceUrl} from '../animationListModule';
 import {changeInterfaceMode} from '../actions';
-import {Goal, show} from '../AnimationPicker/animationPickerModule';
+import {
+  Goal,
+  show,
+  showBackground
+} from '../AnimationPicker/animationPickerModule';
 import i18n from '@cdo/locale';
 import spritelabMsg from '@cdo/spritelab/locale';
-
-function sprites() {
+function animations(backgrounds) {
   const animationList = getStore().getState().animationList;
   if (!animationList || animationList.orderedKeys.length === 0) {
     console.warn('No sprites available');
     return [['sprites missing', 'null']];
   }
-  return animationList.orderedKeys.map(key => {
-    const animation = animationList.propsByKey[key];
-    if (animation.sourceUrl) {
-      return [animation.sourceUrl, `"${animation.name}"`];
-    } else {
-      const url = animationSourceUrl(
-        key,
-        animation,
-        getStore().getState().pageConstants.channelId
-      );
-      return [url, `"${animation.name}"`];
-    }
-  });
+  return animationList.orderedKeys
+    .filter(key => {
+      const animation = animationList.propsByKey[key];
+      if (backgrounds) {
+        return (
+          animation.categories && animation.categories.includes('backgrounds')
+        );
+      } else {
+        return !(
+          animation.categories && animation.categories.includes('backgrounds')
+        );
+      }
```

Comment:
nit: i think we can simplify this filter function as well (definitely double-check my logic here)
```suggestion
      return backgrounds && (animation.categories || []).includes('backgrounds')
```

Reply:
the logic didnt quite work but I tried simplifying it, let me know what you think